src/TypeDefinition.js

```diff
@@ -381,6 +381,7 @@ export type NavigationTransitionProps = {
   // is the index of the scene
   scene: NavigationScene,
   index: number,
+  navigation: NavigationScreenProp<NavigationRoute, NavigationAction>,
```

Comment:
@ericvicenti hard to tell as there are quite a few issues with `flow`. I didn't notice anything relevant. Anyways, regarding the navigation being provided -> we attach it in `_render()` so it's not included between `render()` -> `Transitioner` -> `_render()`.